src/reducers/resetPasswordReducers.js

```diff
@@ -0,0 +1,119 @@
+import types from '../actions/actionTypes';
+
+const { RESET_PASSWORD_LINK, RESET_PASSWORD } = types;
+
+export const initialState = {
+  isLoading: false,
+  success: {
+    status: false,
+    message: '',
+  },
+  error: {
+    status: false,
+    statusCode: 0,
+    message: '',
+  },
+};
+
+const resetPasswordLink = (state = initialState, action = {}) => {
+  switch (action.type) {
+  case `${types.RESET_PASSWORD_LINK}_LOADING`:
+    return {
+      isLoading: true,
+      success: { status: false, message: '' },
+      error: { status: false, message: '' },
+    };
+
+  case `${RESET_PASSWORD_LINK}_SUCCESS`:
+    return {
+      isLoading: false,
+      success: {
+        status: true,
+        message: action.payload.data.message,
+      },
+      error: { status: false },
+    };
+
+  case `${RESET_PASSWORD_LINK}_FAILURE`:
+    return {
+      isLoading: false,
+      success: { status: false },
+      error: {
+        status: true,
+        statusCode: action.payload.statusCode || 0,
+        message: action.payload.message || '',
+      }
+    };
+
+  default:
+    return state;
+  }
+};
+
+
+export const prevState = {
+  token: '',
+  isLoading: false,
+  success: {
+    status: false,
+    message: '',
+  },
+  error: {
+    status: false,
+    statusCode: 0,
+    message: '',
+  },
+};
+
+
+const resetPassword = (state = prevState, action = {}) => {
+  switch (action.type) {
+  case `${RESET_PASSWORD}_TOKEN`:
```

Comment:
why don't you append the `_TOKEN` suffix in action types already since I know the promise middleware won't be doing that for you, except its a success, loading or failure